src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/Referral.kt

```diff
@@ -104,4 +104,7 @@ class Referral(
 
   val currentAssignee: AuthUser?
     get() = currentAssignment?.assignedTo
+
+  val maximumNumberOfSessionsInPreviousAllApprovedPlans: Int?
```

Comment:
i don't think this is nessecary. 

first, we don't need to check all the action plans. the number of sessions can never go down, so we only need to check the most recently approved. 

second, the action plan service can just do `referral.approvedActionPlan.numberOfSessions`